backend/tests.py

```diff
@@ -0,0 +1,66 @@
+import argparse
+from update_db import main as updatedb_main
+from main.server import app, db, models
+from main.server.status import Status
+
+class Arguments:
+    csv_path = None
+    table_name = None
+    dry_run = None
+
+RED="\033[31m"
+GRN="\033[32m"
+YLW="\033[33m"
+CLR="\033[00m"
+
+def main(main_args):
+    # loads a new database into memory. Not persistent
+    app.config.from_object("main.server.config.TestingConfig")
+    args = Arguments()
+    db.create_all()
+
+    # Announcement
+    print(f"{YLW}===RUNNING TESTS FOR ANNOUNCEMENTS==={CLR}")
+    args.csv_path = "./tests/announcement.csv"
+    args.table_name = "ANNOUNCEMENTS"
+    assert updatedb_main(args) == Status.OK
+    assert updatedb_main(args) == Status.WARN
+    print(f"{GRN}[🗸] Test Pass{CLR}")
+
+    # Gallery
+    print(f"{YLW}===RUNNING TESTS FOR GALLERY==={CLR}")
+    args.csv_path = "./tests/gallery.csv"
+    args.table_name = "GALLERY"
+    assert updatedb_main(args) == Status.OK
+    assert updatedb_main(args) == Status.WARN
+    print(f"{GRN}[🗸] Test Pass{CLR}")
+
+    # Games
+    print(f"{YLW}===RUNNING TESTS FOR GAMES==={CLR}")
+    args.csv_path = "./tests/games.csv"
+    args.table_name = "GAMES"
+    assert updatedb_main(args) == Status.OK
+    assert updatedb_main(args) == Status.WARN
+    print(f"{GRN}[🗸] Test Pass{CLR}")
+
+    # Messages
+    print(f"{YLW}===RUNNING TESTS FOR MESSAGES==={CLR}")
+    args.csv_path = "./tests/message.csv"
+    args.table_name = "MESSAGES"
+    assert updatedb_main(args) == Status.OK
+    assert updatedb_main(args) == Status.WARN
+    print(f"{GRN}[🗸] Test Pass{CLR}")
+
+    # Video
+    print(f"{YLW}===RUNNING TESTS FOR VIDEO==={CLR}")
+    args.csv_path = "./tests/video.csv"
+    args.table_name = "VIDEO"
+    assert updatedb_main(args) == Status.OK
+    assert updatedb_main(args) == Status.WARN
+    print(f"{GRN}[🗸] Test Pass{CLR}")
+
+if __name__ == "__main__":
```

Comment:
Good first shot at writing tests. I'd recommend using a unittest style test format, instead of running tests like a straight script.

For example see pytest's intro https://docs.pytest.org/en/6.2.x/

I'd define all the separate calls to whatever as separate tests. Or alternatively, parameterizing everything as tuples of `(csv_path, table_name)`. See https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions for examples on parameterization.

Reply:
Let's leave this as a "to improve" if we need to merge things. At least there's some tests.